tests/test_packages/test_skills_integration/test_simple_aggregation.py

```diff
@@ -0,0 +1,197 @@
+            setting_path = "vendor.fetchai.connections.p2p_libp2p.cert_requests"
+            settings = json.dumps(
+                [
+                    {
+                        "identifier": "acn",
+                        "ledger_id": FetchAICrypto.identifier,
+                        "not_after": "2022-01-01",
+                        "not_before": "2021-01-01",
+                        "public_key": FetchAICrypto.identifier,
+                        "message_format": "{public_key}",
+                        "save_path": ".certs/conn_cert.txt",
+                    }
+                ]
+            )
+            self.set_config(setting_path, settings, type_="list")
+            if i == 0:
+                result = self.run_cli_command(
+                    "get-multiaddress", "fetchai", "--connection", cwd=self._get_cwd()
+                )
+                multiaddr = result.stdout
+            else:
+                setting_path = "vendor.fetchai.connections.p2p_libp2p.config"
+                settings = json.dumps(
+                    {
+                        "delegate_uri": f"127.0.0.1:{11000 + i}",
+                        "entry_peers": [f"/dns4/127.0.0.1/tcp/9000/p2p/{multiaddr}"],
+                        "local_uri": f"127.0.0.1:{9000 + i}",
+                        "log_file": "libp2p_node.log",
+                        "public_uri": f"127.0.0.1:{9000 + i}",
+                    }
+                )
+                self.set_config(setting_path, settings, type_="dict")
+                setting_path = "vendor.fetchai.connections.prometheus.config.port"
+                self.set_config(setting_path, 20000 + i)
+                setting_path = "vendor.fetchai.connections.http_server.config.port"
+                self.set_config(setting_path, 8000 + i)
+
+            # run agent
+            aea_processes.append(self.run_agent())
+
+        for agent in agents:
+            self.set_agent_context(agent)
+            check_strings = (
+                "Starting libp2p node...",
+                "Connecting to libp2p node...",
+                "Successfully connected to libp2p node!",
+                LIBP2P_SUCCESS_MESSAGE,
+                "setting up HttpHandler",
+                "setting up PrometheusHandler",
+                "setting up AdvancedDataRequestBehaviour",
+                "Adding Prometheus metric: num_retrievals",
+                "Adding Prometheus metric: num_requests",
+                "registering agent on SOEF.",
+                "registering service on SOEF.",
+                "Start processing messages...",
+                "Fetching data from",
+                "Observation: {'price': {'value': ",
+                "found agents=",
+                "sending observation to peer=",
+                "received observation from sender=",
+                "Observations:",
+                "Average:",
+            )
+            missing_strings = self.missing_from_output(
```

Comment:
I'd split this up; first make sure all agents launch (so line 167-170), then check for the rest.